### ISSUE TITLE
netbsd tcp() source problem

### ISSUE THREAD

**Issue by czanik** [syslog-ng/syslog-ng#3701]
Working on NetBSD 9.2 I started to update syslog-ng in pkgsrc from version 3.17.2 to 3.32.1. Collecting local logs and over UDP seems to work, but the tcp() source does not. Interestingly enough it shows up in syslog-ng's debug output on startup:
```
[2021-06-10T09:41:16.528278] Module loaded and initialized successfully; module='affile'
[2021-06-10T09:41:16.528457] Running application hooks; hook='6'
[2021-06-10T09:41:16.529528] Accepting connections; addr='AF_INET(172.16.167.136:514)'
[2021-06-10T09:41:16.530396] Running application hooks; hook='1'
[2021-06-10T09:41:16.530493] Running application hooks; hook='7'
```
But it does not show up in the output of netstat and when I try to send logs or just simply telnet to the port, connection is refused:
```
localhost# telnet 172.16.167.136 514
Trying 172.16.167.136...
telnet: Unable to connect to remote host
localhost# 
```
I used the following two patches to get syslog-ng working on NetBSD, updated from pkgsrc:
[patch-lib_signal-handler.txt](https://github.com/syslog-ng/syslog-ng/files/6629362/patch-lib_signal-handler.txt)
[patch-modules_system-source_system-source.txt](https://github.com/syslog-ng/syslog-ng/files/6629363/patch-modules_system-source_system-source.txt)
Without these syslog-ng crashes on startup and system() cannot find local log sources.

**Comment by alltilla** [comment]
signal-handler related issues: #3391 #3445

**Comment by MrAnno** [comment]
We agreed to do the following:
1. We'll merge `patch-modules_system-source_system-source.txt`, so it no longer have to be maintained separately.
2. For the signal-handler issue, we'll check whether we can come up a more portable `sigaction` trickery.
3. @OverOrion is investigating the `network()/tcp()` source issue (`listen()` does not seem to work).

**Comment by OverOrion** [comment]
With the help of the team I was able to identify the root cause: 
It turned out that the `network` source had never worked on NetBSD but the issue manifested only after 4 hours.
Syslog-ng set the `TCP_KEEPALIVE` values on a `listening` socket which causes NetBSD to stop accepting connections after a certain amount of time. On all other platforms this setting did not cause any issues as these values are inherited by the clients.
With syslog-ng-3.29.1 the default values for `TCP_KEEPALIVE` had been changed in #3357. 

We are working on a fix.

**Comment by MrAnno** [comment]
@czanik The TCP source issue is fixed in #3751.

The signal-handler patch needs some additional investigation and discussion, but I think  `patch-modules_system-source_system-source` could be added to the system source. Would you like to open a PR?

**Comment by czanik** [comment]
@OverOrion: thanks, I just tested the tcp() source on NetBSD and seems to work perfectly.

@MrAnno: I'll send a PR soon for system() source. I have to refresh my memories about git first :-)

**Comment by czanik** [comment]
@MrAnno : PR sent: https://github.com/syslog-ng/syslog-ng/pull/3761 A bit slower than expected, as I had to learn GitHub personal access tokens along the way :)

**Comment by MrAnno** [comment]
@czanik Thank you.

**Comment by szemere** [comment]
Fixed by: #3751 and #3761
